odoo/addons/base/tests/test_split_table.py

```diff
@@ -0,0 +1,31 @@
+from odoo.tests.common import TransactionCase
+from odoo.tools import file_open
+from odoo.tools.split_table import split_table
+from lxml import etree
+
+def cleanup_string(s):
+    return ''.join(s.split())
+
+class TestSplitTable(TransactionCase):
+    def test_split_table(self):
+        # NOTE: All the tests's xml are in split_table/ relative to this file
+        CASES = (
+            ("Table's len is equal to max_rows and should not be split", "simple", "simple", 3),
+            ("Table's len is greater to max_rows and should not be split", "simple", "simple", 4),
+            ("max_rows is 1 and every table should be split", "simple", "simple.split1", 1),
+            ("max_row is 2 and the table should be split", "simple", "simple.split2", 2),
+            ("Nested tables should be split", "nested", "nested.split2", 2),
+            ("Nested tables at the start should be split", "first_nested", "first_nested.split2", 2),
+            ("Attributes should be copied", "copy_attributes", "copy_attributes.split1", 1),
+            ("Except the id attribute which shouldn't be copied", "no_copy_id", "no_copy_id.split1", 1),
+        )
+
+        for description, actual, expected, max_rows in CASES:
+            with self.subTest(description), \
+                file_open(f"base/tests/split_table/{expected}.xml") as actual, \
```

Comment:
Be careful about the test, it should be `actual` 😄

Reply:
Well this explains why it worked on the first try :smiling_face_with_tear: